driver/fvGFS/fv_nggps_diag.F90

```diff
@@ -734,6 +743,49 @@ subroutine fv_nggps_diag(Atm, zvir, Time)
       call store_data(id_uhmin25, uhmin25, Time, kstt_uhmin25, kend_uhmin25)
     endif
 
+    IF ( .not. Atm(n)%flagstruct%hydrostatic .and. do_hailcast ) THEN
```

Comment:
It is impossible because it calls _`store_date`_ which is defined in **_fv_nggps_diag.F90_**. Otherwise, it will cause cyclic module dependency. 

We also considered to define a new set of subroutines of _`store_date`_, _`add_field_to_bundle`_, _`find_outputname`_ etc.  inside **_module_diag_hailcast_**. But I do not think having duplicated subroutines for the same function is a good code practice.

Reply:
I understand. This is OK, although something like `store_data` eventually needs to be defined in some better location, or adapted to use standard FMS routines.

Reply:
@lharris - store_data does have take an argument that determines whether the data is handled asynchronously (buffered and sent to the write_component cores) or inline via the FMS diag_manager (send_data).